### PR TITLE
Finish jwt exercise

### PR DESCRIPTION
**Pull request by shaharshw**

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/config/JerseyConfig.kt
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/config/JerseyConfig.kt:1]: revert please

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+@Component
+@Provider
+class AuthenticationFilter : ContainerRequestFilter {
+
+    override fun filter(requestContext: ContainerRequestContext) {
+
+        val path = requestContext.uriInfo.path
+
+        if (path == "api/auth/login") return
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt]: this should be a const

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        val cookies: Map<String, Cookie> = requestContext.cookies
+        val token = cookies["Authorization"]?.value?.trim()
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt]: please make this a const and reuse it in the resource file

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        val claims = validateToken(token)
+
+        if (claims == null) {
+            requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED)
+                .entity("User can't access to the resource")
+                .build())
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt]: do this in the catch

File: src/main/kotlin/com/hibob/academy/resource/AuthenticationResource.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        val cookie = NewCookie.Builder("Authorization")
+            .value(token)
+            .build()
+
+        return Response.ok().cookie(cookie).build()
+    }
+
+    @GET
+    @Path("/check")
+    fun check(): String = "OK"
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/resource/AuthenticationResource.kt]: best practice is to alway return a `Response` object

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import jakarta.ws.rs.container.ContainerRequestContext
+import jakarta.ws.rs.container.ContainerRequestFilter
+import jakarta.ws.rs.core.Cookie
+import jakarta.ws.rs.core.Response
+import jakarta.ws.rs.ext.Provider
+import org.springframework.stereotype.Component
+
+const val PATH_TO_SKIP = "api/auth/login"
+
+const val AUTH_COOKIE_NAME = "Authorization"
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt]: we usually put consts in a companion object. look in the ExampleDao for an example

File: src/main/kotlin/com/hibob/academy/service/SessionService.kt [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+    companion object {
+
+        const val SECRET_KEY: String = "secrethashkeyshahar12323445665713124jadnasdkllFSDFNSD89SLJNLASJNLD"
+    }
+
+    fun createJwtToken(user : LoginUser): String {
+
+        val now : Date = Date.from(LocalDateTime.now().toInstant(ZoneOffset.UTC))
+        val expirationDate : Date = Date.from(LocalDateTime.now().plusDays(1).toInstant(ZoneOffset.UTC))
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/service/SessionService.kt:21]: it's no biggie, but if you already created the now object, you might as well reuse it on this line

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt (hunk 5 above)
Comment (shaharshw) [reply src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt]: got it, thanks

**Review by elijulian — APPROVED** [review]

**Pull request merged**